neo/Network/P2P/Payloads/Cosigner.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class Cosigner : ISerializable
```

Comment:
What about move cosigners to `Transaction`?

```c#
public class Transaction
{
    public byte Version;
    public uint Nonce;
    public UInt160 Sender;
    public long SystemFee;
    public long NetworkFee;
    public uint ValidUntilBlock;
    public TransactionAttribute[] Attributes;
    public Cosigner[] Cosigners;
    public byte[] Script;
    public Witness[] Witnesses { get; set; }
}
```

Reply:
We can remove the sense of Attributes, ValidUntilBlock could be moved inside attributes, for me, TransactionAttribute should be a base class, and extended in CosignerAttribute, this prevent twice deserializations.

Reply:
`ValidUntilBlock` is not an optional field and must be included in every transaction. So it is not suitable to be moved to the attributes.

Reply:
I agree... cosigners is more than attribute to me, its a requirement. Attributes should be more optional, and extendable to the future.
Even url, if we get to launch neo3 without any other idea, we should embeed url and abolish attributes completely.

Reply:
Attributes could be used in the future, like for oracles

Reply:
Then, we can move it outside